src/java.base/share/classes/java/lang/String.java

```diff
@@ -920,7 +922,12 @@ private static byte[] encodeWithEncoder(Charset cs, byte coder, byte[] val, bool
      * Throws iae, instead of replacing, if unmappable.
      */
     static byte[] getBytesUTF8NoRepl(String s) {
-        return encodeUTF8(s.coder(), s.value(), false);
+        byte[] val = s.value();
+        byte coder = s.coder();
+        if (coder == LATIN1 && isASCII(val)) {
```

Comment:
Why don't we just call `s.isLatin1()` instead of `coder == LATIN1 && isASCII(val)`?

Reply:
> Why don't we just call `s.isLatin1()` instead of `coder == LATIN1 && isASCII(val)`?

Here I copied the implementation of `getBytesNoRepl1`.

You are right. The temporary variables here are unnecessary.

Reply:
> Why don't we just call `s.isLatin1()` instead of `coder == LATIN1 && isASCII(val)`?

The `isLatin1()` can only replace `coder == LATIN1`, and it is necessary to check whether it is ASCII.

In the following, we also need to pass the `coder` when calling `encodeUTF8`. Therefore, I think it is clearer to judge the `coder` directly than to call `isLatin1()`.